vitals/ReportGenerator.java

```diff
@@ -0,0 +1,109 @@
+package vitals;
+
+import java.util.Map;
+
+public class ReportGenerator {
```

Comment:
ReportGenerator is a god class .. holds too many responsibility (lack of cohesion)